contributor-docs/adrs/adr-XXX-file-structure.md

```diff
@@ -0,0 +1,37 @@
+# ADR XXX: File structure
+
+## Status
+
+Proposed
+
+## Context
+
+Components might be more grokable if they were structured consistently
+
+## Decision
+
+```
+primer-react/
+├─ src/
+│  ├─ BreadCrumbs/
+│  │  ├─ index.ts                    // Just re-exporting?
+│  │  ├─ BreadCrumbs.tsx             // Primary component
+│  │  ├─ BreadCrumbsItem.tsx         // Subcomponent
+│  │  ├─ BreadCrumbs.mdx             // Documentation. Always .mdx, not .md
+│  │  ├─ BreadCrumbs.stories.tsx
+│  │  ├─ BreadCrumbs.test.tsx
+│  │  └─ BreadCrumbs.types.test.tsx
+┆  ┆
+```
+
+Questions I would like to settle in this ADR:
+
+- On subcomponents:
+  - Do subcomponents filenames need to be namespaced with the parent component? (e.g., `BreadCrumbsItem.tsx` vs `Item.tsx`)
```

Comment:
➕ : Makes searching for related files easier in the IDE

Reply:
Have to say first - super happy to go with what the majority of the team prefers. very loosely help opinions.

Aw, I was going to vote ➖ on this because it is scoped inside the component folder: `src/Breadcrumb/Item.tsx`.

For example, in VS Code, i would search for "action item" to look for `ActionList/Item`

![image](https://user-images.githubusercontent.com/1863771/144937089-455fc4b5-3008-4f1e-8003-e0a116120e78.png)

Reply:
FWIW I prefer the longer, more descriptive file name.

Reply:
FWIW, I prefer the longer more descriptive name.